Polaritonic-Quantum-Chemistry/helper_CS_CQED_CIS.py

```diff
@@ -0,0 +1,326 @@
+    # build the (oo|vv) integrals:
+    oovv = np.asarray(mints.mo_eri(Co, Co, Cv, Cv))
+
+    # strip out occupied orbital energies, eps_o spans 0..ndocc-1
+    eps_o = eps[:ndocc]
+
+    # strip out virtual orbital energies, eps_v spans 0..nvirt-1
+    eps_v = eps[ndocc:]
+
+    # Extra terms for Pauli-Fierz Hamiltonian
+    # nuclear dipole
+    mu_nuc_x = mol.nuclear_dipole()[0]
+    mu_nuc_y = mol.nuclear_dipole()[1]
+    mu_nuc_z = mol.nuclear_dipole()[2]
+
+    # l \cdot \mu_nuc for d_c
+    l_dot_mu_nuc = lambda_vector[0] * mu_nuc_x
+    l_dot_mu_nuc += lambda_vector[1] * mu_nuc_y
+    l_dot_mu_nuc += lambda_vector[2] * mu_nuc_z
+
+    # dipole arrays in AO basis
+    mu_ao_x = np.asarray(mints.ao_dipole()[0])
+    mu_ao_y = np.asarray(mints.ao_dipole()[1])
+    mu_ao_z = np.asarray(mints.ao_dipole()[2])
+
+    # transform dipole array to CQED-RHF basis
+    mu_cmo_x = np.dot(C.T, mu_ao_x).dot(C)
+    mu_cmo_y = np.dot(C.T, mu_ao_y).dot(C)
+    mu_cmo_z = np.dot(C.T, mu_ao_z).dot(C)
+
+    # \lambda \cdot < \mu >
+    # e.g. line 6 of Eq. (18) in [McTague:2021:ChemRxiv]
+    l_dot_mu_exp = 0.0
+    for i in range(0, 3):
+        l_dot_mu_exp += lambda_vector[i] * cqed_rhf_dipole_moment[i]
+
+    # \lambda \cdot \mu_{el}
+    # e.g. line 4 Eq. (18) in [McTague:2021:ChemRxiv]
+    l_dot_mu_el = lambda_vector[0] * mu_cmo_x
+    l_dot_mu_el += lambda_vector[1] * mu_cmo_y
+    l_dot_mu_el += lambda_vector[2] * mu_cmo_z
+
+    # dipole constants to add to E_CQED_CIS,
+    #  0.5 * (\lambda \cdot \mu_{nuc})** 2
+    #      - (\lambda \cdot <\mu> ) ( \lambda \cdot \mu_{nuc})
+    # +0.5 * (\lambda \cdot <\mu>) ** 2
+    # Eq. (14) of [McTague:2021:ChemRxiv]
+    d_c = (
+        0.5 * l_dot_mu_nuc ** 2 - l_dot_mu_nuc * l_dot_mu_exp + 0.5 * l_dot_mu_exp ** 2
+    )
+
+    # check to see if d_c what we have from CQED-RHF calculation
+    assert np.isclose(d_c, cqed_rhf_dict["DIPOLE ENERGY"])
+
+    # create Hamiltonian for elements H[ias, jbt]
+    Htot = np.zeros((ndocc * nvirt * 2 + 2, ndocc * nvirt * 2 + 2), dtype=complex)
+    Hep = np.zeros((ndocc * nvirt * 2 + 2, ndocc * nvirt * 2 + 2), dtype=complex)
+    H1e = np.zeros((ndocc * nvirt * 2 + 2, ndocc * nvirt * 2 + 2), dtype=complex)
+    H2e = np.zeros((ndocc * nvirt * 2 + 2, ndocc * nvirt * 2 + 2), dtype=complex)
+    H2edp = np.zeros((ndocc * nvirt * 2 + 2, ndocc * nvirt * 2 + 2), dtype=complex)
+    Hp = np.zeros((ndocc * nvirt * 2 + 2, ndocc * nvirt * 2 + 2), dtype=complex)
+
+    # elements corresponding to <s|<\Phi_0 | H | \Phi_0>|t>
+    # Eq. (16) of [McTague:2021:ChemRxiv]
+    Hp[0, 0] = 0.0
+    Hp[1, 1] = omega_val
+
+    # elements corresponding to <s|<\Phi_0 | H | \Phi_i^a>|t>
+    # Eq. (17) of [McTague:2021:ChemRxiv]
+    for s in range(0, 2):
+        for i in range(0, ndocc):
+            for a in range(0, nvirt):
+                A = a + ndocc
+                for t in range(0, 2):
+                    iat = 2 * (i * nvirt + a) + t + 2
+                    Hep[s, iat] = (
```

Comment:
```suggestion
                    Hep[s, iat] = Hep[iat, s] = (
```

We shouldn't duplicate the transpose.